### PR TITLE
Make table listings sorted in output HTML

### PR DESCRIPTION
**Pull request by DarkWiiPlayer**

### CODE REVIEW

**Comment by daelvn** [comment]
I trust that this is tested and it works Fine:tm:?

**Comment by DarkWiiPlayer** [comment]
I haven't encountered any problems with it at least so it should™ be fine

File: template.moonxml [hunk 1]
```diff
@@ -55,8 +55,22 @@ dump = (k=ROOT, v) ->
...
+                a,b = a[1], b[1]
+                if type a < type b then
+                  return true
+                elseif type a > type b then
+                  return false
+                else
+                  if type a == "number"
+                    return a < b
+                  else
+                    return tostring a < tostring b
```
Comment (daelvn) [inline template.moonxml]: Unfortunately, MoonScript associativity SUCKS hard. Have you looked at the generated code?
```lua
              if type(a < type(b)) then
                return true
              elseif type(a > type(b)) then
                return false
              else
                if type(a == "number") then
                  return a < b
                else
                  return tostring(a < tostring(b))
                end
              end
```
Parentheses are unavoidable in some cases unfortunately.
```suggestion
                if (type a) < (type b) then
                  return true
                elseif (type a) > (type b) then
                  return false
                else
                  if (type a) == "number"
                    return a < b
                  else
                    return (tostring a) < (tostring b)
```

File: template.moonxml.lua [hunk 2]
```diff
@@ -53,8 +53,29 @@ dump = function(k, v)
...
+              a, b = a[1], b[1]
+              if type(a < type(b)) then
+                return true
+              elseif type(a > type(b)) then
+                return false
+              else
+                if type(a == "number") then
+                  return a < b
+                else
+                  return tostring(a < tostring(b))
```
Comment (daelvn) [inline template.moonxml.lua]: Manually fix the generated code. Yes, I know, horrible idea, but I know what I'm doing.
```suggestion
              if type(a) < type(b) then
                return true
              elseif type(a) > type(b) then
                return false
              else
                if type(a) == "number" then
                  return a < b
                else
                  return tostring(a) < tostring(b)
```

**Review by daelvn — CHANGES_REQUESTED** [review]

**Review by daelvn — APPROVED** [review]

**Pull request merged**